### ISSUE TITLE
update to v0.25.0 leads to "Uncaught TypeError: Cannot call a class as a function"

### ISSUE THREAD

**Issue by barbalex** [acdlite/recompose#488]
When I update from v0.24.0 to v0.25.0 my app breaks with above error.

Here is the output in the console:

    main.9d27d69f.js:72780 Uncaught TypeError: Cannot call a class as a function
        at r (main.9d27d69f.js:72780)
        at t (main.9d27d69f.js:72780)
        at a (createEagerElementUtil.js:16)
        at createEagerFactory.js:18
        at u.render (withState.js:60)
        at f._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:798)
        at f._renderValidatedComponent (ReactCompositeComponent.js:821)
        at f.performInitialMount (ReactCompositeComponent.js:361)
        at f.mountComponent (ReactCompositeComponent.js:257)
        at Object.mountComponent (ReactReconciler.js:45)
    r	@	main.9d27d69f.js:72780
    t	@	main.9d27d69f.js:72780
    a	@	createEagerElementUtil.js:16
    (anonymous)	@	createEagerFactory.js:18
    u.render	@	withState.js:60
    _renderValidatedComponentWithoutOwnerOrContext	@	ReactCompositeComponent.js:798
    _renderValidatedComponent	@	ReactCompositeComponent.js:821
    performInitialMount	@	ReactCompositeComponent.js:361
    mountComponent	@	ReactCompositeComponent.js:257
    mountComponent	@	ReactReconciler.js:45
    mountChildren	@	ReactMultiChild.js:236
    _createInitialChildren	@	ReactDOMComponent.js:703
    mountComponent	@	ReactDOMComponent.js:522
    mountComponent	@	ReactReconciler.js:45
    performInitialMount	@	ReactCompositeComponent.js:370
    mountComponent	@	ReactCompositeComponent.js:257
    mountComponent	@	ReactReconciler.js:45
    performInitialMount	@	ReactCompositeComponent.js:370
    mountComponent	@	ReactCompositeComponent.js:257
    mountComponent	@	ReactReconciler.js:45
    performInitialMount	@	ReactCompositeComponent.js:370
    mountComponent	@	ReactCompositeComponent.js:257
    mountComponent	@	ReactReconciler.js:45
    performInitialMount	@	ReactCompositeComponent.js:370
    mountComponent	@	ReactCompositeComponent.js:257
    mountComponent	@	ReactReconciler.js:45
    performInitialMount	@	ReactCompositeComponent.js:370
    mountComponent	@	ReactCompositeComponent.js:257
    mountComponent	@	ReactReconciler.js:45
    performInitialMount	@	ReactCompositeComponent.js:370
    mountComponent	@	ReactCompositeComponent.js:257
    mountComponent	@	ReactReconciler.js:45
    a	@	ReactMount.js:104
    perform	@	Transaction.js:143
    s	@	ReactMount.js:126
    perform	@	Transaction.js:143
    batchedUpdates	@	ReactDefaultBatchingStrategy.js:62
    i	@	ReactUpdates.js:97
    _renderNewRootComponent	@	ReactMount.js:319
    _renderSubtreeIntoContainer	@	ReactMount.js:401
    render	@	ReactMount.js:422
    (anonymous)	@	index.js:68
    t	@	bootstrap 212a8f5…:49
    (anonymous)	@	main.9d27d69f.js:38090
    t	@	bootstrap 212a8f5…:49
    (anonymous)	@	bootstrap 212a8f5…:144
    (anonymous)	@	bootstrap 212a8f5…:144

This only happens in production, not in dev mode.

Here is my app: https://github.com/barbalex/brexit

**Comment by oliviertassinari** [comment]
This issue could be linked to #473. (I still don't get why the change was made).

**Comment by istarkov** [comment]
Duplicate of #489

**Comment by barbalex** [comment]
I changed the `render = () =>` calls now and hope that production will work.

If I remember right, it is not necessary (as in other functions inside a class) because render binds `this` (please correct me if this is wrong).
But is there anything saying that binding a render function is _wrong_?

And I feel that this behavior is _very_ bad as it breaks an app _only_ in production. It really bit me hard.

**Comment by istarkov** [comment]
We had the same issue in the past but with more rare case. As I wrote in related issue it will be good to provide additional checks at dev mode, it can be a good first PR ;-)
About render, Having that there is no documented way to check that something is react class we used undocumented method, and in latest release changed to 'duck typing' check method, both are bad, but latest allows users of Preact and Inferno to use recompose. We are open to better ideas.
